src/codegen/mod.rs

```diff
@@ -3962,6 +3964,53 @@ impl CodeGenerator for ObjCInterface {
                         }
                     };
                     result.push(impl_trait);
+                    for protocol_id in parent.conforms_to.iter() {
+                        if protocol_set.insert(*protocol_id) {
+                            let protocol_name = ctx.rust_ident(
+                                ctx.resolve_type(
+                                    protocol_id.expect_type_id(ctx),
+                                )
+                                .name()
+                                .unwrap(),
+                            );
+                            let impl_trait = quote! {
+                                impl #protocol_name for #class_name { }
+                            };
+                            result.push(impl_trait);
+                        }
+                    }
+                    if !parent.is_template() {
+                        let parent_struct_name = parent.name();
+                        let child_struct_name = self.name();
+                        let parent_struct = ctx.rust_ident(parent_struct_name);
+                        let from_block = quote! {
+                            impl From<#class_name> for #parent_struct {
+                                fn from(child: #class_name) -> #parent_struct {
+                                    #parent_struct(child.0)
+                                }
+                            }
+                        };
+                        result.push(from_block);
+
+                        let error_msg = format!(
+                            "This {} cannot be downcasted to {}",
+                            parent_struct_name, child_struct_name
+                        );
+                        let try_into_block = quote! {
+                            impl std::convert::TryFrom<#parent_struct> for #class_name {
```

Comment:
I think this can't use `std` if the `no_std` flag is set, so this should use `trait_prefix`. Also, probably should use the old style root paths (`::std::convert::TryFrom`) for consistency with other code and to support Rust 2015.